tests/test_coverity.py

```diff
@@ -1,126 +1,236 @@
+
+    def test_retrieve_columns(self):
+        with open(f"{TEST_FOLDER}/columns_keys.json", "r") as content:
+            column_keys = json.loads(content.read())
+        self.fake_stream = "test_stream"
+        # initialize what needed for the REST API
+        coverity_service = self.initialize_coverity_service(login=True)
+        with requests_mock.mock() as mocker:
+            mocker.get(self.column_keys_url, json=column_keys)
+            coverity_service.retrieve_column_keys()
+            assert mocker.call_count == 1
+            mock_request = mocker.last_request
+            assert mock_request.method == "GET"
+            assert mock_request.url == self.column_keys_url
+            assert mock_request.verify
+            assert coverity_service.columns["Issue Kind"] == "displayIssueKind"
+            assert coverity_service.columns["CID"] == "cid"
+
+    def test_retrieve_checkers(self):
+        self.fake_stream = "test_stream"
+        self.fake_checkers = {
+            "checkerAttribute": {"name": "checker", "displayName": "Checker"},
+            "checkerAttributedata": [
+                {"key": "MISRA", "value": "MISRA"},
+                {"key": "CHECKER", "value": "CHECKER"}
+            ],
+        }
+        # initialize what needed for the REST API
+        coverity_service = self.initialize_coverity_service(login=True)
+
+        with requests_mock.mock() as mocker:
+            mocker.get(self.checkers_url, json=self.fake_checkers)
+            coverity_service.retrieve_checkers()
+            history = mocker.request_history
+            assert mocker.call_count == 1
+            assert history[0].method == "GET"
+            assert history[0].url == self.checkers_url
+            assert history[0].verify
+            assert coverity_service.checkers == ["MISRA", "CHECKER"]
+
+    @parameterized.expand([
+        [test_defect_filter_0.filters, test_defect_filter_0.column_names, test_defect_filter_0.request_data],
+        [test_defect_filter_1.filters, test_defect_filter_1.column_names, test_defect_filter_1.request_data],
+        [test_defect_filter_2.filters, test_defect_filter_2.column_names, test_defect_filter_2.request_data],
+        [test_defect_filter_3.filters, test_defect_filter_3.column_names, test_defect_filter_3.request_data]
+    ])
+    def test_get_defects(self, filters, column_names, request_data):
+        with open(f"{TEST_FOLDER}/columns_keys.json", "r") as content:
+            column_keys = json.loads(content.read())
+        self.fake_checkers = {
+            "checkerAttribute": {"name": "checker", "displayName": "Checker"},
+            "checkerAttributedata": [
+                {"key": "MISRA 1", "value": "MISRA 1"},
+                {"key": "MISRA 2", "value": "MISRA 2"},
+                {"key": "MISRA 3", "value": "MISRA 3"},
+                {"key": "CHECKER 1", "value": "CHECKER 1"},
+                {"key": "CHECKER 2", "value": "CHECKER 2"}
```

Comment:
Why did you choose the exact same dummy values for both key and value? You're not testing whether key or value are used correctly.

Reply:
Because they are the same, but I will change them to test it correctly

Reply:
Resolved in bfd8fbf3a26fa53e55408883acdd5f2b201569d5